rules/TypeDeclaration/Rector/ClassMethod/AddVoidReturnTypeWhereNoReturnRector.php

```diff
@@ -140,4 +143,40 @@ private function changePhpDocToVoidIfNotNever(ClassMethod|Function_|Closure|Node
 
         $this->phpDocTypeChanger->changeReturnType($phpDocInfo, new VoidType());
     }
+
+    private function shouldSkipClassMethod(ClassMethod|Function_|Closure $functionLike): bool
+    {
+        if (! $functionLike instanceof ClassMethod) {
+            return false;
+        }
+
+        if ($functionLike->isMagic()) {
+            return true;
+        }
+
+        if ($functionLike->isAbstract()) {
+            return true;
+        }
+
+        if ($functionLike->isProtected()) {
+            return ! $this->isInsideFinalClass($functionLike);
+        }
```

Comment:
@TomasVotruba this should use `ClassChildAnalyzer` service to detect has child class method also, `VendorLockResolver` service for vendor detection, also, if `protected` is going to be skipped, then `public` as well, ref https://3v4l.org/3J6TO

Reply:
I see. 

This case was breaking on purpose protected method in one project. It belong to similar category like `abstract` method.

I'd stick with public methods here, as it would make rule much less usable. E.g. all test method that shoud have `void` would be skipped etc.

Reply:
At first sight, it seems good. Could you add it + test fixtures too?

Reply:
I will try.